configure

```diff
@@ -1942,851 +1943,526 @@ webos)
 	;;
 esac
 
+
 #
-# Determine the C++ compiler
+# Check the CPU architecture
 #
-echo_n "Looking for C++ compiler... "
```

Comment:
this mostly moving host detection before compiler detection

Reply:
In that case, can you split this into two separate commits so that the changes made are easier to understand?

Reply:
I second that, this particular commit is way too big and not clear at all what is going on.

Reply:
I split the change to separate commit, and this change is only doing that - moving a huge chunk of code around, sadly diff is too intelligent and it's displaying it like this

Reply:
just moving everything between
```
#
# Check the CPU architecture
#
```
and 
```
#
# Backend related stuff
#
```

just before 
```
#
# Determine the C++ compiler
#
```

Reply:
this part was removed from this change